kgb-unlock.c

```diff
@@ -10,6 +9,14 @@
 
 #include <unistd.h>
 
+char *mememem(char *haystack, size_t haystack_len, char *needle, size_t needle_len) {
+    for (size_t i = 0; i < haystack_len; i++) {
+        if (i + needle_len > haystack_len) return NULL;
+        if (strncmp(&haystack[i], needle, needle_len) == 0) return &haystack[i];
```

Comment:
I think it would suit better to use `memcmp` here since we are working with generic memory instead of null-terminated strings.

In the use case below needle is indeed null-terminated, but I'd consider that a pure accident.

Reply:
You make a good point, since we don't need additional checks on null termination from strncmp we'll (in theory) gain performance by using a less strict version of that algorithm. Changed code accordingly.